doc/design-component.md

```diff
@@ -0,0 +1,69 @@
+need to keep the Operator running to keep the API objects in the desired states. On the other hand, Operators' all-on pattern does not serve MicroShift's small footprint design goal.
+
+## Kubernetes Addon
+Kubernetes Addon mechanism loads cluster wide infrastructure after cluster initialization. Addons such as networking, storage, and management are curated in the Kubernetes repo. Such arrangement, however, does not fit 
+into the MicroShift use case:
+- MicroShift may need more than one components. For instance, Ceph bucket notification, a use case for processing
+Object Store objects on demand, involves Rook/Ceph and Knative (Eventing, Serving, and Networking). 
+- These components are often inter-dependent. Kubernetes Addons cannot ensure the components are created or started in the desired order.
+
+## Helm
+Helm packages multiple components in a chart form and deploy them in a declarative manner. Once deployed, Helm exists till the next it upgrades or removes the chart from the cluster. The non-interactive and low footprint features fit into MicroShift use well. Still the chart must be accessible by the MicroShift cluster when operating
+in an airgapped environment.
+
+# Design
+As discussed, MicroShift Component Loader must be able to:
+- Customizable for different workloads targeting varying use cases
+- Managed multiple, sometimes inter-dependent component
+- Allow MicroShift to invoke when the cluster is ready
+- Allow MicroShift to monitor readiness status
+- Low, preferrably zero footprint
+- Able to operate in offline mode
+
+## Component Loader Workflow
+
+When MicroShift starts up the cluster, it reads from the configuration file and finds the Container image that Component Load Job (i.e. running as a Kubernetes Job) uses, constructs a Kubernetes Job using this Container images, creates one ConfigMap for the Job to report progress, creates another ConfigMap for parameters required by the Job (if exists), then starts and watches the Job.
+
+When the Job starts, it reads its configuration from the parameter ConfigMap, creates components, reports readiness status by updating the predefined ConfigMap.
+
+Once the Job exits, MicroShift catches the status and checks the ConfigMap and gets the status.
+
+# Issues
+
+## Repair Components
+It is likely that after startup, some of the components, such as Pods and Deployments, degrade or are even killed. To cope with this case, the Component Loader can be started as a CronJob so it can start periodically to repair the components.
+
+## Offline Operation Support
+
+### Container Image
+The Container images of the Component Loader and the compnonents it created are packaged or pre-downloaded when integrated with MicroShift.
+
+### Reuse Helm Chart
+Similarly, if Helm Charts are used as or by a Component Loader, the Charts are also locally stored together with the airgapped Container images.
```

Comment:
Locally stored, but maybe in different directories? For example, `/var/lib/microshift/images` vs `/var/lib/microshift/charts`?